src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/RenderMode_Directive_FullyQualified/TestComponent.codegen.cs

```diff
@@ -0,0 +1,58 @@
+﻿// <auto-generated/>
+#pragma warning disable 1591
+namespace Test
+{
+    #line hidden
+    using global::System;
+    using global::System.Collections.Generic;
+    using global::System.Linq;
+    using global::System.Threading.Tasks;
+    using global::Microsoft.AspNetCore.Components;
+    [global::Test.PrivateComponentRenderModeAttribute]
+    public partial class TestComponent : global::Microsoft.AspNetCore.Components.ComponentBase
+    {
+        #pragma warning disable 219
+        private void __RazorDirectiveTokenHelpers__() {
+        ((global::System.Action)(() => {
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+global::System.Object __typeHelper = nameof(Microsoft.AspNetCore.Components.DefaultRenderModes.Server);
+
+#line default
+#line hidden
+#nullable disable
+        }
+        ))();
+        }
+        #pragma warning restore 219
+        #pragma warning disable 0414
+        private static object __o = null;
+        #pragma warning restore 0414
+        #pragma warning disable 1998
+        protected override void BuildRenderTree(global::Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder)
+        {
+        }
+        #pragma warning restore 1998
+    }
+    file sealed class PrivateComponentRenderModeAttribute : global::Microsoft.AspNetCore.Components.RenderModeAttribute
```

Comment:
Btw, what if C# LangVersion is set to 10 and RazorLangVersion to 8. Is that supported? Because that would fail:

```cs
[Fact]
public async Task RenderMode_CSharp10()
{
    // Arrange
    var project = CreateTestProject(new()
    {
        ["Shared/Component1.razor"] = """
            @rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
            """,
    }, new()
    {
        ["RenderModes.cs"] = """
            using System;
            namespace Microsoft.AspNetCore.Components;
            
            public interface IComponentRenderMode
            {
            }
            
            [AttributeUsage(AttributeTargets.Class)]
            public abstract class RenderModeAttribute : Attribute
            {
                public abstract IComponentRenderMode Mode { get; }
            }
            
            public class DefaultRenderModes : IComponentRenderMode
            {
               public static IComponentRenderMode Server = new DefaultRenderModes();
            }
            """,
    });
    project = project.WithParseOptions(((CSharpParseOptions)project.ParseOptions).WithLanguageVersion(LanguageVersion.CSharp10));
    var compilation = await project.GetCompilationAsync();
    var driver = await GetDriverAsync(project);

    // Act
    var result = RunGenerator(compilation!, ref driver,
        // Microsoft.NET.Sdk.Razor.SourceGenerators\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator\Shared_Component1_razor.g.cs(21,23): error CS8936: Feature 'file types' is not available in C# 10.0. Please use language version 11.0 or greater.
        //     file sealed class PrivateComponentRenderModeAttribute : global::Microsoft.AspNetCore.Components.RenderModeAttribute
        Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion10, "PrivateComponentRenderModeAttribute").WithArguments("file types", "11.0").WithLocation(21, 23));

    // Assert
    Assert.Empty(result.Diagnostics);
    Assert.Single(result.GeneratedSources);
}
```